src/webui/www/private/scripts/mocha-init.js

```diff
@@ -745,29 +776,32 @@ const initializeWindows = function() {
     deleteTorrentsByCategoryFN = function(categoryHash) {
         const hashes = torrentsTable.getFilteredTorrentsHashes("all", categoryHash, TAGS_ALL, TRACKERS_ALL);
         if (hashes.length > 0) {
-            window.qBittorrent.Client.closeWindow("confirmDeletionPage");
-
-            new MochaUI.Window({
-                id: "confirmDeletionPage",
-                icon: "images/qbittorrent-tray.svg",
-                title: "QBT_TR(Remove torrent(s))QBT_TR[CONTEXT=confirmDeletionDlg]",
-                data: { hashes: hashes },
-                loadMethod: "xhr",
-                contentURL: "views/confirmdeletion.html",
-                maximizable: false,
-                collapsible: false,
-                padding: {
-                    left: 5,
-                    right: 10,
-                    top: 15,
-                    bottom: 15
-                },
-                width: 480,
-                onContentLoaded: function(w) {
-                    MochaUI.resizeWindow(w, { centered: true });
-                    MochaUI.centerWindow(w);
-                }
-            });
+            if (LocalPreferences.get("confirm_torrent_deletion", "true") === "true") {
```

Comment:
It was necessary to copy and paste it multiple times here and below in order to make confirm option work in all cases and to keep existing code intact. Currently, WebUI does not properly take into account selected filters (and filter terms) when performing "Start/stop/remove torrents" context menu actions. I'm going to improve this behavior in another PR and most of the repeated code from here will be gone. I just wanted to do things in proper order.